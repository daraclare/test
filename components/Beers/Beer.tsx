import React from "react";
import styled from "styled-components";
import Link from "next/link";

const BeerWrapper = styled.div`
  padding: 16px;
`;

const StyledDiv = styled.div`
  display: flex;
`;

const StyledImg = styled.img`
  padding: 20px 20px 0 0px;
`;

const StyledP = styled.p`
  overflow: scroll;
  text-overflow: ellipsis;
`;

const StyledH4 = styled.h4`
  position: absolute;
  margin: 5px;
  bottom: 0;
  right: 10px;
`;

const StyledA = styled.a`
  color: unset;
  text-decoration: none;
`;

export const Beer = ({ id, name, img, description, abv }) => {
  return (
    <BeerWrapper>
      <Link href={`/${id}`}>
        <StyledA>
          <h2 data-testid="beer-name">{name}</h2>
          <StyledDiv>
            <StyledImg
              data-testid="beer-img"
              src={img}
              alt={`beer label`}
              height="100px"
            />
            <StyledP data-testid="beer-desc">{description}</StyledP>
          </StyledDiv>
          <StyledH4 data-testid="beer-abv">ABV: {abv}</StyledH4>
        </StyledA>
      </Link>
    </BeerWrapper>
  );
};
