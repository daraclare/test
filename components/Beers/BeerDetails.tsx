import React, { FC } from "react";
import styled from "styled-components";
import { BeerType } from "../../types/types";

type BeerDetailsProps = { beer: BeerType };

const StyledDiv = styled.div``;

export const BeerDetails: FC<BeerDetailsProps> = ({ beer }) => {
  console.log("beer", beer);
  const {
    abv,
    description,
    food_pairing,
    id,
    image_url,
    ingredients: { hops, malt, yeast },
    method: { fermentation, mash_temp },
    name,
    tagline
  } = beer;
  return (
    <StyledDiv>
      <h2>
        {name} <span>({tagline})</span>
      </h2>
      <p>{abv}</p>
      <p>{description}</p>
      <h4>Food Pairings: </h4>
      {food_pairing.map((food: string, index: number) => (
        <p key={index}>{food}</p>
      ))}
      <img src={image_url} alt={`bottle of ${name}`} />
    </StyledDiv>
  );
};
