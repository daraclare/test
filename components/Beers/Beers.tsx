import React from "react";
import styled from "styled-components";
import { Beer } from "./Beer";

const StyledArticle = styled.article`
  margin: 10px;
  border: 1px solid lightgrey;
  border-radius: 16px;
  position: relative;
  &:hover {
    border: 1px solid darkslategrey;
  }
`;

export const Beers = ({ beers }) => {
  return beers.map((beer: any) => {
    const { abv, description, id, name, image_url } = beer;
    return (
      <StyledArticle data-testid="beer" key={`beer-${id}`}>
        <Beer
          data-testid={`beer-${id}`}
          name={name}
          img={image_url}
          description={description}
          abv={abv}
          id={id}
        />
      </StyledArticle>
    );
  });
};
