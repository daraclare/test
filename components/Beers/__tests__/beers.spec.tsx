import React from "react";
import { render } from "@testing-library/react";
import { Beers } from "../Beers";
import { BEERS_MOCK } from "../__mocks__/beers.mocks";

describe("Component: Beers", () => {
  it("should render without throwing an error", () => {
    const { container } = render(<Beers beers={BEERS_MOCK} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render with correct number of beers", () => {
    const { getAllByTestId } = render(<Beers beers={BEERS_MOCK} />);
    const beers = getAllByTestId("beer");
    expect(beers).toHaveLength(7);
  });
});
