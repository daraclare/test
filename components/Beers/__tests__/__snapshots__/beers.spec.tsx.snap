// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Beers should render without throwing an error 1`] = `
<article
  class="sc-bZQynM dTBqDA"
  data-testid="beer"
>
  <div
    class="sc-bdVaJa eIxvgw"
  >
    <a
      class="sc-EHOje hYajMM"
    >
      <h2
        data-testid="beer-name"
      >
        Buzz
      </h2>
      <div
        class="sc-bwzfXH jIEWCi"
      >
        <img
          alt="beer label"
          class="sc-htpNat cBsyOg"
          data-testid="beer-img"
          height="100px"
          src="https://images.punkapi.com/v2/keg.png"
        />
        <p
          class="sc-bxivhb byyMnK"
          data-testid="beer-desc"
        >
          A light, crisp and bitter IPA brewed with English and American hops. A small batch brewed only once.
        </p>
      </div>
      <h4
        class="sc-ifAKCX cycDOG"
        data-testid="beer-abv"
      >
        ABV: 
        4.5
      </h4>
    </a>
  </div>
</article>
`;
